keanu-project/src/main/java/io/improbable/keanu/algorithms/particlefiltering/ParticleFilter.java

```diff
@@ -132,7 +134,7 @@ private void addObservedVertexToParticles(List<Particle> particles,
     }
 
     private <T> void sampleValueAndAddToParticle(Vertex<T> vertex, Particle particle) {
-        T sample = vertex.sample(random);
+        T sample = ((Probabilistic<T>) vertex).sample(random);
```

Comment:
This looks unsafe but the `vertex` is coming from a set inside `sort` which extracts only latent variables